### PR TITLE
fix: additional listener in graphic widgets states for appropriate modules loading

### PR DESCRIPTION
**Pull request by KirylHatalski**
Ticket: <a href="https://oat-sa.atlassian.net/browse/AUT-2903">AUT-2903</a>

## What's Changed

- New custom event was created for align required rendering modules loading in scope of figure usage.

## How to test
- Need to add image with inline alignment into the prompt field on any graphic interaction test, the next loading of this test should be without templating error.

### CODE REVIEW

**Comment by jsconan** [comment]
It looks good. Maybe could you try to limit the duplication of code, but it is not a hard stop. If the fix works, you can move forward.

**Review by marpesia — CHANGES_REQUESTED** [review]
Good job. It fixes the issue!

Please add the code in these interactions too:
- graphicAssociateInteraction
- graphicGapMatchInteraction

File: views/js/qtiCreator/widgets/helpers/responsiveMetaChange.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            }
+            if($container.find('.qti-prompt [data-qti-class="figure"]').length) {
+                $container.on('graphicInteraction.ready', function (){
+                    widget.rebuild();
+                });
+                return;
+            }
+            widget.rebuild();
+        }
+    }
```
Comment (jsconan) [inline views/js/qtiCreator/widgets/helpers/responsiveMetaChange.js:40]: **praise:** The factorization looks nice!

**chore:** New code should be written in ES6+ whenever possible.

**Review by jsconan — COMMENTED** [review]
The approach looks good. Nice job!

Could you write the new code in ES6+?

**Comment by marpesia** [comment]
It is working well in the four cases.
@jsconan do you think the helper file is ok on the widget/helpers path, or it could be better on widget/interactions/helpers path?

File: views/js/qtiCreator/widgets/helpers/responsiveMetaChange.js (hunk 1 above)
Comment (KirylHatalski) [reply views/js/qtiCreator/widgets/helpers/responsiveMetaChange.js]: Done

**Review by marpesia — APPROVED** [review]
An excellent solution to the Figure loading problem!

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by jsconan — APPROVED** [review]
It looks good. Tested locally.

Before merging, please make sure to apply Prettier, the new code does not respect our code style.

image

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code respects code style rules
- [x] New code respects best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by viktar-dzmitryieu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`

**Review by jsconan — APPROVED** [review]

**Pull request merged**